managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/OpReadEntry.java

```diff
@@ -100,8 +100,8 @@ public void readEntriesFailed(ManagedLedgerException exception, Object ctx) {
         } else if (cursor.config.isAutoSkipNonRecoverableData() && exception instanceof NonRecoverableLedgerException) {
             log.warn("[{}][{}] read failed from ledger at position:{} : {}", cursor.ledger.getName(), cursor.getName(),
                     readPosition, exception.getMessage());
-            // try to find and move to next valid ledger
-            final Position nexReadPosition = cursor.getNextLedgerPosition(readPosition.getLedgerId());
+            // Skip this read operation
+            PositionImpl nexReadPosition = cursor.ledger.getValidPositionAfterSkippedEntries(readPosition, count);
```

Comment:
Please do not access a field, use a getter

Reply:
OK , I will fixed!

Reply:
fixed

Reply:
Fixed!  PTAL,thanks! @eolivelli